src/api/app/models/kiwi/image.rb

```diff
@@ -141,11 +141,38 @@ def outdated?
   end
 
   def default_package_group
-    package_groups.find_or_create_by(kiwi_type: :image, pattern_type: 'onlyRequired')
+    package_groups.find_or_create_by(kiwi_type: :image)
+  end
+
+  def find_binaries_by_name(query)
+    finder = /\A#{Regexp.quote(query)}/
+    binaries_available.select { |package, _| finder.match(package.to_s) }
+  end
+
+  def binaries_available
+    return {} unless package
+    Rails.cache.fetch("kiwi_image_binaries_available_#{md5_last_revision}", expires_in: 5.minutes) do
+      if use_project_repositories
+        Backend::Api::BuildResults::Binaries.available_in_project(package.project.name)
+      else
+        url_repos = repositories.reject(&:obs_source_path?).map(&:source_path)
+        obs_repos = repositories.select(&:obs_source_path?).map { |rep| "#{rep.project_for_type_obs}/#{rep.repository_for_type_obs}" }
+        url_binaries = Backend::Api::BuildResults::Binaries.available_in_urls(package.project.name, url_repos)
+        obs_binaries = Backend::Api::BuildResults::Binaries.available_in_repositories(package.project.name, obs_repos)
+        merge_binaries(url_binaries, obs_binaries)
```

Comment:
See my comments in my previous review, you can combine the call and don't need to merge.